src/libraries/System.Private.CoreLib/src/System/Reflection/RuntimeMethodBody.cs

```diff
@@ -0,0 +1,25 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+
+namespace System.Reflection
+{
+    internal sealed partial class RuntimeMethodBody : MethodBody
+    {
+        private byte[] _IL = null!;
+        private ExceptionHandlingClause[] _exceptionHandlingClauses = null!;
+        private LocalVariableInfo[] _localVariables = null!;
+        internal MethodBase _methodBase = null!;
```

Comment:
Should this field be in CoreCLR version of the type only?

Reply:
It was failing an assertion in checked build because of the native definition (field offset) mismatch. I'll surround it with `#if CORECLR` here.

Reply:
It would be better to change CoreCLR to the same plan as Mono and call constructor from the runtime.

Reply:
That would be nice, but it is a non-trivial work. Mono computes the values and calls the ctor once and then does not make changes to the object from runtime side. Coreclr does make changes (reassign references) to these objects right after calling the ctor, and also later elsewhere in the runtime.

There are other types in System/Reflection, where both runtimes make changes to the object fields after instantiation. I moved this type as it mainly differ by the ctor.